_spinnaker_install_admin_guides/secrets-vault.md

```diff
@@ -0,0 +1,115 @@
+---
+layout: post
+title: Secrets with Vault
+order: 152
+---
+
+{:toc}
+
+
+This document describes how to set up Spinnaker secrets in Hashicorp's Vault. In this example, we'll be using the default kv secret engine called `secret` and will be storing github credentials, a kubeconfig file and a java keystore for SAML SSO.
+
+
+## Authorization
+
+We currently support two methods of authenticating to your Vault server.
+
+### 1. Kubernetes service account (recommended)
+
+You'll need to configure Vault to authenticate with Kubernetes per [Hashicorp's documentation](https://www.vaultproject.io/docs/auth/kubernetes.html#configuration). 
+
+Note: If multiple clusters need to access the same Vault server, you'll need to use the [-path flag](https://www.vaultproject.io/docs/commands/auth/enable.html#usage) and give each cluster a different path name. This becomes `<cluster auth path>` in the example below. If using just one cluster, you can use the default `vault auth enable kubernetes` command, in which case your path will be `kubernetes`.
+
+```yaml
+secrets:
+  vault:
+    enabled: true
+    url: <Vault server URL>
+    authMethod: KUBERNETES
+    role: <k8s role with access to Vault>
+    path: <cluster auth path>
+```
+
+### 2. Token authentication
+
+This method is not recommended but it is supported if you choose. For token authentication, you'll need to have a `VAULT_TOKEN` environment variable set for halyard and each of the services.
+
+```yaml
+secrets:
+  vault:
+    enabled: true
+    url: <Vault server URL>
+    authMethod: TOKEN
+```
+
+## Configuring Halyard to use Vault secrets
+The halyard daemon will need access to the Vault server in order to decrypt within halyard for validation, etc., so you'll also need to add the config block to your `halyard.yml`:
+
+```yaml
+secrets:
+  vault:
+    enabled: true
+    url: <Vault server URL>
+    authMethod: KUBERNETES
+    role: <k8s role>
+    path: <k8s cluster path>
+```
+Then restart the daemon (you'll only need to do this the first time):
+```
+hal shutdown
+```
+Your next hal command will automatically bring the daemon back up.
```

Comment:
this is only true when installed locally. Wonder if we need to distinguish b/w that case and in a Docker container.